sympy/matrices/common.py

```diff
@@ -2379,7 +2389,7 @@ class MatrixArithmetic(MatrixRequired):
     def _eval_Abs(self):
         return self._new(self.rows, self.cols, lambda i, j: Abs(self[i, j]))
 
-    def _eval_add(self, other):
+    def _eval_matrix_add(self, other):
         return self._new(self.rows, self.cols,
```

Comment:
I think that this is the place where the unified domain of `self` and `other` should be added as the `ring` parameter to `_new`?